### DISCUSSION TITLE
pyenv bin file has errors

### DISCUSSION THREAD

**Discussion by EldrinSMP** [pyenv/pyenv#2921] [General] [answered]
I first noticed there were problems with the pyenv file when I first tried to run `eval "$(pyenv init -)"` and got the error:

> pyenv: no such command `init'

Naturally, my next step was to use `pyenv help` which returnedf 

> pyenv: no such command 'help'

The same thing happend for the `pyenv commands` command.  So I tried `pyenv` by itself and got back 

> /bin/pyenv: line 110: pyenv---version: command not found

This is now suggesting that there was an issue with the build. I used the PYENV-Installer command to install pyenv, and I used the development environment for my distro (SOLUS) from the wiki. The post install environment is in place. I'm at a loss as to what's going on.  Anyone have any ideas? I'm not sure if this qualifies as something to put in "Issues" as of yet.

Your help would be greatly appreciated.

**Comment by native-api** [comment, marked as answer]
> /bin/pyenv: line 110: pyenv---version: command not found

This is not normal. Where did you install Pyenv and where did this file come from? The `pyenv` executable on PATH is supposed to be a symlink to the same-named file in the `libexec` directory where all the other commands reside.

**Comment by EldrinSMP** [comment]
> > /bin/pyenv: line 110: pyenv---version: command not found
> 
> This is not normal. Where did you install Pyenv and where did this file come from? The `pyenv` executable on PATH is supposed to be a symlink to the same-named file in the `libexec` directory where all the other commands reside.

As far as I know, it was created when I ran `curl https://pyenv.run | bash` to install pyenv. I have purged that installation from my system and was actually about to try again. I'll let you know how it goes.

**Comment by saylerb** [comment]
I had a similar issue on MacOS and the problem was that the symlink at `~/.pyenv/bin/pyenv` was not created correctly.

OS: MacOS 15.3.2
Git: git version 2.39.5 (Apple Git-154)

```
ls -l ~/.pyenv/bin/pyenv
-rw-r--r--@ 1 $USER  staff  16 Apr 16 17:47 ~/.pyenv/bin/pyenv
```

I found that `core.symlinks` was set to `false` in my git config (you can check with `git config --get core.symlinks`,[ see documentation](https://git-scm.com/docs/git-config#Documentation/git-config.txt-coresymlinks)).

It appears that I temporarily set this [due to a vulnerability with Git](https://github.com/git/git/security/advisories/GHSA-8h77-4q3w-gfgv) (good thing I left a comment in my `~/.gitconfig`) around a year ago. 

Can fix this with
```
git config --global core.symlinks true
```

Then uninstall and reinstall pyenv:
```
rm -rf ~/.pyenv
curl https://pyenv.run | bash
```

Obviously, ensure `git` is up to date to make sure you have a version that has patched that vulnerability.
